src/Energon.php

```diff
@@ -37,11 +37,7 @@ public static function generate(int $prime = null, int $size = Optimus::DEFAULT_
 
     public static function generatePrime(int $size = Optimus::DEFAULT_SIZE): int
     {
-        $max = self::createMaxInt($size);
-        $expForMin = max(1, floor(log10($max->toString())) - 2);
-        $min = new BigInteger(10 ** $expForMin);
-
-        return (int) $max->randomPrime($min, $max)->toString();
+        return (int) BigInteger::randomPrime($size)->toString();
```

Comment:
I think there's a possible bug here. I remember there was a reason why I picked a specific minimum. This change would allow 0 to size.

Reply:
The `$size` is a bit length. You did convert from size to range because phpseclib2 haven`t method with size argument. Now phpseclib3 have method with size argument and do the same conversion. Please check the [BigInteger::randomPrime](https://github.com/phpseclib/phpseclib/blob/a127a5133804ff2f47ae629dd529b129da616ad7/phpseclib/Math/BigInteger/Engines/Engine.php#L690) 
 source code.